### PR TITLE
Pass exit code to command launcher

### LINKED ISSUE #46: 🚸 Hide exit code
The cdt command outputs `Error: exit status 1` for some cases, let's hide this

### PR DESCRIPTION
**Pull request by bhou-crto**
Also hide the exit code message at the end of execution

close #46

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -369,22 +371,22 @@ func getExecutableCommand(group, name string) (command.Command, error) {
 }
 
 // execute a cdt command
-func executeCommand(group, name string, args []string) error {
+func executeCommand(group, name string, args []string) (int, error) {
 	iCmd, err := getExecutableCommand(group, name)
 	if err != nil {
-		return err
+		return 404, err
```
Comment (emmanuelguerin) [inline cmd/root.go:377]: This doesn't look like standard Nix error code but rather http ones?

File: cmd/root.go [hunk 2]
```diff
@@ -236,13 +236,13 @@ func addCommands(groups []command.Command, executables []command.Command) {
...
 			Short:              v.ShortDescription(),
 			Long:               v.LongDescription(),
-			RunE: func(cmd *cobra.Command, args []string) error {
-				err := executeCommand(group, name, args)
+			Run: func(cmd *cobra.Command, args []string) {
+				exitCode, err := executeCommand(group, name, args)
 				if err != nil && err.Error() == EXECUTABLE_NOT_DEFINED {
 					cmd.Help()
-					return nil
+					os.Exit(exitCode)
```
Comment (emmanuelguerin) [inline cmd/root.go:243]: I'm not too familiar with the architecture of the code, but shouldn't the actual exit done by the caller to the run method?

**Review by emmanuelguerin — CHANGES_REQUESTED** [review]
I don't see any tests about the propagation of the exit code?

File: cmd/root.go (hunk 2 above)
Comment (bhou-crto) [reply cmd/root.go:243]: You are right, this could skip the postRun process. I will make a patch on it

File: cmd/root.go (hunk 1 above)
Comment (bhou-crto) [reply cmd/root.go]: Fixed in new commit

**Review by lhutyra — APPROVED** [review]

**Pull request merged**